CMakeLists.txt

```diff
@@ -1,40 +1,24 @@
-cmake_minimum_required(VERSION 2.6)
+cmake_minimum_required(VERSION 3.0)
+project(fcitx5-fbterm)
 
-project(fcitx-fbterm)
+set(CMAKE_CXX_STANDARD 14)
 
 find_package(PkgConfig REQUIRED)
-find_package(Gettext REQUIRED)
-find_package(Fcitx 4.2.6 REQUIRED)
+find_package(Fcitx5Core REQUIRED)
+find_package(Fcitx5GClient REQUIRED)
 
-pkg_check_modules(FCITX_GCLIENT "fcitx-gclient>=4.2.6" REQUIRED)
+pkg_check_modules(GIO "gio-2.0" REQUIRED)
 
-# uninstall target
-configure_file(
-    "${CMAKE_CURRENT_SOURCE_DIR}/cmake/cmake_uninstall.cmake.in"
-    "${CMAKE_CURRENT_BINARY_DIR}/cmake_uninstall.cmake"
-    IMMEDIATE @ONLY)
+include("${FCITX_INSTALL_CMAKECONFIG_DIR}/Fcitx5Utils/Fcitx5CompilerSettings.cmake")
 
-add_custom_target(uninstall
-    COMMAND ${CMAKE_COMMAND} -P ${CMAKE_CURRENT_BINARY_DIR}/cmake_uninstall.cmake)
+add_executable(fcitx5-fbterm fcitx5-fbterm.cpp imapi.c imapi.c immessage.h keycode.cpp keycode.h input_key.h keymap.cpp keymap.h utils.h utils.cpp fcitx5-fbterm.h)
 
-if(NOT DEFINED LIB_INSTALL_DIR)
-    set(LIB_INSTALL_DIR ${CMAKE_INSTALL_PREFIX}/lib)
-endif()
+target_include_directories(fcitx5-fbterm PRIVATE
+        ${CMAKE_CURRENT_SOURCE_DIR}
+        ${CMAKE_CURRENT_BINARY_DIR}
+        ${FCITX5_CORE_INCLUDE_DIRS}
+        ${FCITX5_UTILS_INCLUDE_DIRS}
+        ${FCITX5_GCLIENT_INCLUDE_DIRS}
```

Comment:
You can also remove these lines, as long as you use Fcitx5::Core. It will be added automatically.